### PR TITLE
snaprepoapi: Add Region field and WithRegion ctx

### PR DESCRIPTION
**Pull request by karencfv**
## Description
This patch modifies all the snaprepoapi public functions to have a
Region string required parameter which is then used to create a new
context.Context: api.WithRegion(context.Background(), params.Region)
and pass it as the API operation context.

Unit tests have been modified to include a request
assertion which ensures that the region path interpolation is used when
the /platform APIs are being called.

The repository.go, repository_params.go and repository_test.go files have 
been split and reorganised into multiple files each containing a single 
API context and its parameters.

## Related Issues
Related: #129 

## How Has This Been Tested?
unit tests in ecctl

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Readiness Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by karencfv** [comment]
I could not for the love of god figure out why the `TestSet` test is failing with:
```console
* got body {"settings":{"region":"us-east-1","bucket":"mybucket","access_key":"myaccesskey","secret_key":"mysecretkey"},"type":"s3"}
    , want {"settings":{"region":"us-east-1","bucket":"mybucket","access_key":"myaccesskey","secret_key":"mysecretkey"},"type":"s3"}
```

😭 

@marclop, do you have any ideas on why this is happening?

**Comment by marclop** [comment]
@karencfv ah yes, I've run across this, you need to add `+"\n"` to the body you're generating sadly.

File: pkg/api/platformapi/snaprepoapi/set_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+						Response: http.Response{
+							StatusCode: 200,
+							Body:       mock.NewStringBody(`{}`),
+						},
+						Assert: &mock.RequestAssertion{
+							Header: api.DefaultWriteMockHeaders,
+							Method: "PUT",
+							Host:   api.DefaultMockHost,
+							Path:   "/api/v1/regions/us-east-1/platform/configuration/snapshots/repositories/snapshot_repo_name",
+							Body:   mock.NewStringBody(`{"settings":{"region":"us-east-1","bucket":"mybucket","access_key":"myaccesskey","secret_key":"mysecretkey"},"type":"s3"}`),
```
Comment (marclop) [inline pkg/api/platformapi/snaprepoapi/set_test.go:65]:
```suggestion
							Body:   mock.NewStringBody(`{"settings":{"region":"us-east-1","bucket":"mybucket","access_key":"myaccesskey","secret_key":"mysecretkey"},"type":"s3"}`+ "\n"),
```

**Comment by karencfv** [comment]
@marclop thank you!

**Pull request merged**